includes/class-dispatcher.php

```diff
@@ -125,25 +125,33 @@ public function _ajax_dispatch() {
 
     public function get_view_locations() {
         $dirs = array();
-        $dirs[] = WPBDP_PATH . 'includes/views/';
+        $dirs[] = WPBDP_PATH . 'includes/controllers/pages/';
         $dirs[] = WPBDP_PATH . 'core/views/';
 
         return apply_filters( 'wpbdp_view_locations', $dirs );
     }
 
     public function load_view( $view_name, $args = null ) {
+        $class_view_name = str_replace( '_', '-', $view_name );
         // TODO: add some filters so plugins can override default view loading.
         $filenames = array( $view_name . '.php',
-                            'views-' . $view_name . '.php' );
-
+                            'views-' . $view_name . '.php',
+                            'class-' . $class_view_name . '.php',
+                        );
+        $find = array( '.php', 'class' );
         foreach ( $this->get_view_locations() as $dir ) {
             foreach ( $filenames as $f ) {
                 $path = wp_normalize_path( WPBDP_FS::join( $dir, $f ) );
 
                 if ( ! file_exists( $path ) )
                     continue;
 
-                $classname = 'WPBDP__Views__' . implode( '_', array_map( 'ucfirst', explode( '_', str_replace( '.php', '', $f ) ) ) );
+                // Views that start with class- in the file name. Addons may not have this hence the condition.
+                if ( strpos( $f, 'class' ) !== false ) {
```

Comment:
This would be a bit safer here:
`strpos( $f, 'class' ) === 0`

Reply:
Issue of checking for `0` as per the docs:

```
Returns false if the needle was not found.
This function may return Boolean false, but may also return a non-Boolean value which evaluates to false. Please read the section on Booleans for more information. Use the === operator for testing the return value of this function.
```

Keeping it as false is safer here as it may return something else

Reply:
It returns the position in the string when it's found. In this case, we are only looking for class at the beginning of the string.